src/mem/heap/heap_simple.c

```diff
@@ -134,6 +134,48 @@ void *malloc(size_t size) {
 	return 0;
 }
 
+void *memalign(size_t boundary, size_t size) {
+	void *ptr;
+	struct mem_control_block *mcb, *next_mcb;
+
+	if (size == 0 || (boundary & (boundary - 1)) == 1) {
```

Comment:
What does `(boundary & (boundary - 1)) == 1` mean? It just verifies that `boundary` is odd, isn't it?

Reply:
But the boundary is required to be a power of two.

Reply:
It exactly checks if it is power of 2. I added one fix in my second commit and I check if it is different from 0.

Reply:
>  I added one fix in my second commit and I check if it is different from 0.

Yes, it's right.